controllers/front/product.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+use LupaSearch\LupaSearchPlugin\AuthorizationValidator;
+use LupaSearch\LupaSearchPlugin\ProductDataProvider;
+use LupaSearch\LupaSearchPlugin\QueryProvider;
+use Symfony\Component\HttpFoundation\Request;
+
+class LupaSearchProductModuleFrontController extends ModuleFrontController
+{
+    private $authorizationValidator;
+    private $productDataProvider;
+    private $queryProvider;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->authorizationValidator = new AuthorizationValidator();
+        $this->queryProvider = new QueryProvider();
+        $this->productDataProvider = new ProductDataProvider($this->queryProvider);
+    }
+
+    public function initContent(): void
+    {
+        parent::initContent();
+
+        try {
+            $this->authorizationValidator->validateRequest(Request::createFromGlobals());
+        } catch (Throwable $e) {
+            header('Content-Type: application/json');
+            header('HTTP/1.1 403 Forbidden');
+            $this->ajaxRender(json_encode($e->getMessage()));
+            die();
```

Comment:
```suggestion
            $this->send($e->getMessage(), AjaxReponse::HTTP_FORBIDDEN);
```

`
class AjaxReponse extends ModuleFrontController {
    const HTTP_OK = 200;
    const HTTP_FORBIDDEN = 400;
    const HTTP_TEXT = [
       self::HTTP_OK => 'OK',
       self::HTTP_FORBIDDEN => 'Forbidden'
   ];

   public function send($content, $code): void
   {
            header('Content-Type: application/json');
            header("HTTP/1.1 $code " . self::HTTP_TEXT[$code]);
            $this->ajaxRender(json_encode($content,));
            exit();
    }
}
`